include/OpenSpaceToolkit/Physics/Environment/Utility/Eclipse.hpp

```diff
@@ -19,20 +26,133 @@ namespace utilities
 {
 
 using ostk::core::container::Array;
+using ostk::core::type::Real;
+using ostk::core::type::String;
+
+using ostk::mathematics::object::Vector3d;
 
 using ostk::physics::coordinate::Position;
 using ostk::physics::Environment;
+using ostk::physics::environment::object::Celestial;
+using ostk::physics::time::Duration;
+using ostk::physics::time::Instant;
 using ostk::physics::time::Interval;
 
+/// @brief                      Phase of an eclipse
+class EclipsePhase
+{
+   public:
+    /// @brief                  Type of an eclipse phase
+    enum class Region
+    {
+        UMBRA,    ///< Umbra region
+        PENUMBRA  ///< Penumbra region
+    };
+
+    Region region;      ///< The region of the eclipse phase
+    Interval interval;  ///< The time interval of the phase
+    bool is_complete;   ///< Whether the phase is complete
+
+    /// @brief                  Constructor
+    ///
+    /// @param                  [in] aRegion The region of the eclipse phase
+    /// @param                  [in] anInterval The time interval of the phase
+    /// @param                  [in] anIsComplete Whether the phase is complete
+    EclipsePhase(const Region& aRegion, const Interval& anInterval, const bool& anIsComplete);
+
+    /// @brief                  Get the region as string
+    ///
+    /// @return                 String representation of the region
+    String getRegionString() const;
+};
+
+/// @brief                      Eclipse
+class Eclipse
+{
+   public:
+    /// @brief                  Constructor
+    ///
+    /// @param                  [in] anOccultedCelestialObject The occulted celestial object
+    /// @param                  [in] anOccultingCelestialObject The occulting celestial object
+    /// @param                  [in] somePhases The phases of the eclipse
+    ///
+    /// @throw                  std::runtime_error If the phases are not contiguous
+    Eclipse(
+        const Celestial& anOccultedCelestialObject,
+        const Celestial& anOccultingCelestialObject,
+        const Array<EclipsePhase>& somePhases
+    );
+
+    /// @brief                  Get the occulted celestial object
+    ///
+    /// @return                 The occulted celestial object
+    const Celestial& getOccultedCelestialObject() const;
+
+    /// @brief                  Get the occulting celestial object
+    ///
+    /// @return                 The occulting celestial object
+    const Celestial& getOccultingCelestialObject() const;
```

Comment:
Since we're returning references, this should be access

```suggestion
    /// @brief                  Access the occulted celestial object
    ///
    /// @return                 The occulted celestial object
    const Celestial& accessOccultedCelestialObject() const;

    /// @brief                  Access the occulting celestial object
    ///
    /// @return                 The occulting celestial object
    const Celestial& accessOccultingCelestialObject() const;
```